lib/utils/utils.ts

```diff
@@ -18,15 +25,41 @@ export const getExternalIp = () => {
         resolve(body);
       });
       res.on('error', (err: Error) => {
-        reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+        reject(p2pErrors.EXTERNAL_IP_UNRETRIEVABLE(err));
       });
 
     }).on('error', (err: Error) => {
-      reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+      reject(p2pErrors.EXTERNAL_IP_UNRETRIEVABLE(err));
     });
   });
 };
 
+/**
+ * Creates an URI from the public key, host and port
+ */
+export const getUri = (uriParts: UriParts): string => {
+  const { pubKey, host, port } = uriParts;
+  return `${pubKey}@${host}:${port}`;
+};
+
+/**
+ * Splits an URI into the public key, host and port
+ */
+export const parseUri = (uri: string): UriParts => {
+  // A regex that splits the string by the symbols "@" and ":"
+  const split = uri.split(/[@:]+/);
+
+  if (split.length !== 3) {
+    throw serviceErrors.INVALID_ARGUMENT('uri is inavlid');
```

Comment:
Yeah I thought that we should handle this specific case because if the uri doesn't contain an `@` there are just two entries in the array `split` which results in the port being `NaN` and that would show the error `port must be an integer between 1024 and 65535`. And this error message isn't the actual problem so I thought that we should check if the uri itself is valid first.

Reply:
Ah you mean like if someone does a uri of `127.0.0.1:8885` it would report the `port must be an integer between 1024 and 65535` error which isn't exactly the case.

How about this, simply do an `assert(split.length === 3, 'uri is invalid')` and then from service do a try/catch when parsing the uri and `throw serviceErrors.INVALID_ARGUMENT(err.message)` from the catch block. That way the logic that's specific to the Service class - namely this INVALID_ARGUMENT error - is not in the general `utils` class should we ever use this method elsewhere. 

Also I just noticed that the message here says "inavlid" instead of "invalid" so fix the typo if you copy/paste the message.